### PR TITLE
Add workflows for Surface Go and Surface Pro 6

### PR DESCRIPTION
**Pull request by klutchell**
- [x] Create PR with all devices supported in the [balena-<device repo> multi job in Jenkins](https://jenkins.product-os.io/view/Yocto/job/yocto-balena-intel/)
- [ ] Merge this PR and check that it deployed to staging on push event
- [ ] Open a new PR to set deploy environment to `balena-cloud.com`
- [ ] Once all builds and tests are passing, disable the associated multi job in Jenkins
- [ ] Update branch protection to require the GHA workflows

See https://balena.fibery.io/Work/Improvement/Make-a-checklist-to-mark-a-repository-BTD-certified-1965

### CODE REVIEW

File: .github/workflows/genericx86-64.yml [hunk 1]
```diff
@@ -1,31 +1,62 @@
...
+        type: boolean
+        default: false
 
 jobs:
   yocto:
     name: Yocto
-    # FIXME: This workflow has dependencies on scripts in the balena-yocto-scripts repository
-    # which is pinned separately as a submodule in the device repo. Expect some drift but try to retain compatibility.
-    uses: balena-os/balena-yocto-scripts/.github/workflows/yocto-build-deploy.yml@v1.25.3
+    uses: balena-os/balena-yocto-scripts/.github/workflows/yocto-build-deploy.yml@7b65acc0280a2239d7f8fcfd7260151d97b6f3d5 # v1.25.9
```
Comment (vipulgupta2048) [inline .github/workflows/genericx86-64.yml:59]: Would renovate be handling the update of the Yocto BTD workflow, I see you have update the workflow manually which is fine as this is the first PR. Getting clarication.

Comment (klutchell) [reply .github/workflows/genericx86-64.yml:59]: Yes, Renovate is already opening PRs to bump the workflow version (the first commit of this PR is actually cherry-picked from the Renovate PR).

It's just the Renovate PR doesn't always merge for the usual reasons of Jenkins, etc.

Comment (vipulgupta2048) [reply .github/workflows/genericx86-64.yml:59]: Yes, that's where my question. Why not commit the latest version here? `v1.25.9`

Comment (klutchell) [reply .github/workflows/genericx86-64.yml:59]: Oversight, I didn't notice there was a newer version available.

**Review by rcooke-warwick — APPROVED** [review]

**Pull request merged**